pkg/operator/controller/gatewayapi/controller.go

```diff
@@ -114,40 +100,3 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	return reconcile.Result{}, nil
 }
-
-// featureIsEnabled takes a feature name and a featuregate config API object and
-// returns a Boolean indicating whether the named feature is enabled.
-//
-// This function determines whether a named feature is enabled as follows:
-//
-//   - First, if the featuregate's spec.featureGateSelection.featureSet field is
-//     set to "CustomNoUpgrade", then the feature is enabled if, and only if, it
-//     is specified in spec.featureGateSelection.customNoUpgrade.enabled.
-//
-//   - Second, if spec.featureGateSelection.featureSet is set to a value that
-//     isn't defined in configv1.FeatureSets, then the feature is *not* enabled.
-//
-//   - Finally, the feature is enabled if, and only if, the feature is specified
-//     in configv1.FeatureSets[spec.featureGateSelection.featureSet].enabled.
```

Comment:
Could this comment be preserved somewhere?  It took me a bit of effort to  work out from the features API documentation and Slack conversations what the logic is for enabling a feature.

Reply:
> Could this comment be preserved somewhere? It took me a bit of effort to work out from the features API documentation and Slack conversations what the logic is for enabling a feature.

Spoke on slack.  With the just-today-merged FeatureGate.Status, this comment is no longer pertinent.  We'll update in two stages so ingress isn't a guinea pig, but we won't preserve this comment.

Reply:
@miciah I use `featureIsEnabled` in my E2E tests and can preserve it there.